lasy/utils/laser_utils.py

```diff
@@ -589,11 +583,11 @@ def field_to_envelope(grid, dim, phase_unwrap_nd=False):
     omg_h, omg0_h = get_frequency(
         grid,
         dim=dim,
-        is_envelope=False,
         is_hilbert=True,
         phase_unwrap_nd=phase_unwrap_nd,
     )
     field *= np.exp(1j * omg0_h * grid.axes[-1])
+    grid.is_envelope = True
```

Comment:
I think that we also need to change the data type of `grid.temporal_field` to  `complex128`.
If not, we get a `complex` to `float` conversion warning in `grid.set_temporal_field`.
```
ComplexWarning: Casting complex values to real discards the imaginary part
  self.temporal_field[:, :, :] = field
 ```